src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/UpdateLiveVacancyCommandHandler.cs

```diff
@@ -33,15 +35,36 @@ public async Task Handle(UpdateLiveVacancyCommand message, CancellationToken can
         {
             _logger.LogInformation("Updating live vacancy {vacancyId}.", message.Vacancy.Id);
 
+            Vacancy originalVacancy = await _repository.GetVacancyAsync(message.Vacancy.Id);
+
             message.Vacancy.LastUpdatedDate = _timeProvider.Now;
             message.Vacancy.LastUpdatedByUser = message.User;
 
             await _repository.UpdateAsync(message.Vacancy);
 
+            await PublishLiveVacancyClosingDateChangedEvent(originalVacancy, message.Vacancy.ClosingDate);
+
             await _messaging.PublishEvent(new VacancyPublishedEvent
             {
                 VacancyId = message.Vacancy.Id
             });
         }
+
+        private async Task PublishLiveVacancyClosingDateChangedEvent(Vacancy originalVacancy, DateTime? newClosingDate)
+        {
+            bool shouldPublishEvent =
+                originalVacancy.Status == VacancyStatus.Live
```

Comment:
You shouldn't need to inspect the original vacancy status as this CommandHandler should only be operating on a Live vacancies. If you are concerned about this then it is better to do the check for this in the `Handle` method.